lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -428,7 +428,7 @@ private function getAssociationValue(ResultSetMapping $rsm, $assocAlias, $entity
      * @param mixed        $value
      * @param array<mixed> $path
      *
-     * @return array<object>|object|null
```

Comment:
Why do you widen the type here? Was this wrong?

Reply:
Yes, or at least the recursivity line 456 makes it hard to understand for psalm.

At level 6, you get `ERROR: InvalidReturnStatement - lib/Doctrine/ORM/Cache/DefaultQueryCache.php:459:16 - The inferred type 'list<array<array-key, object>|null|object>' does not match the declared return type 'array<array-
key, object>|null|object' for Doctrine\ORM\Cache\DefaultQueryCache::getAssociationPathValue (see http
s://psalm.dev/128)`.